tokens-filter/src/lib.rs

```diff
@@ -0,0 +1,280 @@
+use std::sync::Arc;
+
+use anyhow::Ok;
+use async_trait::async_trait;
+use coingecko_client::CoingeckoClient;
+use coinmarketcap_client::CoinmarketcapClient;
+use http_client::settings::HttpClientSettings;
+use permissions_list::PermissionsList;
+use solana_client::nonblocking::rpc_client::RpcClient;
+use solana_sdk::pubkey::Pubkey;
+
+use crate::{json::JsonChecker, jupiter::JupiterChecker};
+
+pub mod coingecko;
+pub mod coinmarketcap;
+pub mod json;
+pub mod jupiter;
+pub mod permissions_list;
+pub mod solana;
+
+pub enum Checker {
+    Json(JsonChecker),
+    Coinmarketcap(CoinmarketcapClient),
+    Coingecko(CoingeckoClient),
+    Jupiter(JupiterChecker),
+    Solana(Arc<RpcClient>),
+}
+
+impl std::fmt::Display for Checker {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let msg = match self {
+            Checker::Json(_) => "Json",
+            Checker::Coinmarketcap(_) => "Coinmarketcap",
+            Checker::Coingecko(_) => "Coingecko",
+            Checker::Jupiter(_) => "Jupiter",
+            Checker::Solana(_) => "Solana",
+        };
+
+        f.write_str(msg)
+    }
+}
+
+#[async_trait]
+impl CheckToken for Checker {
+    type Token = Pubkey;
+
+    async fn check_token(&self, token: &Self::Token) -> anyhow::Result<bool> {
+        match self {
+            Checker::Json(x) => x.check_token(token),
+            Checker::Coinmarketcap(x) => x.check_token(token),
+            Checker::Coingecko(x) => x.check_token(token),
+            Checker::Jupiter(x) => x.check_token(token),
+            Checker::Solana(x) => x.check_token(token),
+        }
+        .await
+    }
+}
+
+impl From<JsonChecker> for Checker {
```

Comment:
You can use [`derive_more`](https://crates.io/crates/derive_more) crate to derive these impls.